runtime/v2/shim/shim.go

```diff
@@ -281,7 +281,7 @@ func NewShimClient(ctx context.Context, svc shimapi.TaskService, signals chan os
 }
 
 // Serve the shim server
-func (s *Client) Serve() error {
+func (s *Client) Serve(cancel context.CancelFunc) error {
```

Comment:
Wondering. `cancel` is also passed to `initFunc`, but only used as a local variable there; https://github.com/containerd/containerd/blob/218db0f9afaeda2813bddbddf2764c8414bb9c42/runtime/v2/runc/v2/service.go#L103

Would it make sense to add the cancelfunc as a property on the service, and have the service itself call it on `Shutdown()` ?

Reply:
🤦 ah, you're right, I somehow missed it was already added as a property. Trying to find what the `ShutdownRequest.Now` property is used for (thinking: if this is because we want it to shut down, could we use that property to decide wether or not `cancel()` must be called even if there's containers running). 

I see it was added in https://github.com/containerd/containerd/commit/da1b5470cd26e9e2b4b1cd0100c876ce8db87471, but at a glance I cannot see any place it's used (all not in the current code); what that property meant for things like this?

Reply:
Yeah I pretty much found the same thing, `Now` is never used.

Reply:
@crosbymichael do you recall what it was added for? Should it be used for this purpose?

Reply:
yeah it feels odd to pass the cancel to both the initFunc and then to the client Serve func, .. feels like we need a get for the client's previously stored s.service.cancel field

Reply:
Yes, so perhaps `if ShutdownRequest.Now == true`, then always call `s.cancel()` (inside `service.ShutDown()`)